qos-service-impl/src/main/java/com/palantir/atlasdb/qos/com/palantir/atlasdb/qos/agent/QosAgent.java

```diff
@@ -32,6 +36,21 @@ public QosAgent(Supplier<QosServiceRuntimeConfig> config, Consumer<Object> regis
     }
 
     public void createAndRegisterResources() {
-        registrar.accept(new QosResource(config));
+        QosClientConfigLoader qosClientConfigLoader = QosClientConfigLoader.create(() -> config.get().clientLimits());
+        ClientLimitMultiplier clientLimitMultiplier = getNonLiveReloadableClientLimitMultiplier();
+        registrar.accept(new QosResource(qosClientConfigLoader, clientLimitMultiplier));
+    }
+
+    private ClientLimitMultiplier getNonLiveReloadableClientLimitMultiplier() {
+        if (config.get().qosCassandraMetricsConfig().isPresent()) {
```

Comment:
the mixing of having lambdas but not supporting live reload is confusing - are we planning to support live reload in a future PR?

Reply:
getting rid of the suppliers for now. Whether we want to support live-reloading here can be a question for another day.